customer-config/checkout-config/che.plulist.v2.json

```diff
@@ -0,0 +1,221 @@
+      "id": {
+        "description": "A unique identifier for the PLU list.",
+        "type": "string",
+        "minLength": 1,
+        "maxLength": 20,
+        "example": "top-5-items-plu"
+      },
+      "name": {
+        "description": "The PLU List name",
+        "type": "string",
+        "minLength": 1,
+        "maxLength": 40,
+        "example": "PLU List"
+      },
+      "description": {
+        "description": "The PLU List description",
+        "type": "string",
+        "minLength": 1,
+        "maxLength": 255,
+        "example": "The description for PLU List"
+      },
+      "deviceTypes": {
+        "description": "A list of device types that this PLUList applies to",
+        "type": "array",
+        "example": "[{'id': 'SelfCheckout'}]",
+        "items": {
+          "type": "object",
+          "properties": {
+            "id": {
+              "type": "string",
+              "minLength": 1,
+              "maxLength": 40
+            }
+          }
+          }
+      },
+      "isModifiable": {
+        "description": "If it is possible to modify the content of this PLUList further down the hierarchy",
+        "type": "boolean",
+        "example": "true"
+      },
+      "isRemovable": {
+        "description": "If it is possible to remove the content of this PLUList further down the hierarchy",
+        "type": "boolean",
+        "example": "true"
+      },
+      "content": {
+        "description": "PLU lists can contain other PLU lists and/or items.",
+        "type": "array",
+        "items": {
+          "$ref": "#/$defs/PluContentType"
+        }
+      },
+      "additionalProperties": {
+        "description": "Additional custom properties can be added via this field.",
+        "type": "array",
+        "items": {
+          "$ref": "#/$defs/AdditionalProperty"
+        }
+      }
+    }
+  },
+  "$defs": {
+    "PluContentType": {
+      "type": "object",
+      "oneOf": [
+        {
+          "$ref": "#/$defs/PluItemType"
+        },
+        {
+          "$ref": "#/$defs/PluGroupType"
+        }
+      ]
+    },
+    "PluItemType": {
+      "description": "This is an entry for a specific item, and is placed inside a PLU list.",
+      "type": "object",
+      "required": [
+        "identifier",
+        "images",
+        "type"
+      ],
+      "properties": {
+        "images": {
+          "$ref": "#/$defs/ImagesType"
+        },
+        "identifier": {
+          "description": "The, item identifier reference",
+          "type": "string",
+          "minLength": 1,
+          "maxLength": 20
+        },
+        "type": {
+          "description": "The, ITEM type is for PLU items.",
+          "type": "string",
+          "enum": [
+            "ITEM"
+          ]
+        }
+      }
+    },
+    "PluGroupType": {
+      "description": "This is a PLU - it contains a set of other PLUs to form a hierarchy.",
+      "type": "object",
+      "required": [
+        "id",
+        "images",
```

Comment:
Should the `images` property be required?
We don't have support for it on the UI yet. I would consider making it optional

Reply:
also lets make group content required

Reply:
as well as name

Reply:
Changed